056-zk-kraft-migration.md

```diff
@@ -0,0 +1,247 @@
+# ZooKeeper to KRaft migration
+
+This proposal is about enabling the Strimzi Cluster Operator to migrate a ZooKeeper-based Apache Kafka cluster to be KRaft-based.
+The Strimzi Cluster Operator would allow the user to drive the migration process across multiple phases in a semi-automatic procedure.
+
+## Current situation
+
+Currently, the Strimzi Cluster Operator supports the deployment of KRaft-based Apache Kafka clusters only from scratch.
+The KRaft support is still behind a `UseKraft` feature gate which also needs the `KafkaNodePools` one to work correctly.
+Through the usage of one or more `KafkaNodePool` custom resources, alongside the main `Kafka` one, the user is able to define pools of brokers and KRaft controllers, taking ZooKeeper out of the picture.
+
+It is not possible to migrate a ZooKeeper-based Apache Kafka cluster to be KRaft based.
+If the user has an already deployed Apache Kafka cluster using a ZooKeeper ensemble for storing metadata, the Strimzi Cluster Operator has no support to migrate the running cluster to use a pool of KRaft based controllers replacing the ZooKeeper nodes for the same purpose.
+
+There is also no support to upgrade a KRaft-based Apache Kafka cluster to a newer version, but this missing piece is out of the scope for this proposal.
+
+## Motivation
+
+The KRaft mode has been declared to be production-ready since Apache Kafka 3.3 release.
```

Comment:
I would leave this out or specify by whom it was declared production-ready. Because we do not consider it production ready.

Reply:
This statement is in the KIP-833 (I have put the link a few lines below). This KIP was discussed and accepted by different Kafka maintainers. I could leave this out or saying "by the Apache Kafka community/maintainers". @mimaison @showuon @tombentley any opinions on this?

Reply:
Yes you can add `by the Apache Kafka project` if you want to clarify